src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSignedCommits.java

```diff
@@ -0,0 +1,204 @@
+
+  /**
+   * A GitHub token provided by user to access the GitHub API.
+   */
+  private final String githubToken;
+
+  /**
+   * The date format to be used during GitHub API call.
+   */
+  private static final SimpleDateFormat DATE_FORMAT =
+      new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.US);
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param where A GitHub organization or user name.
+   * @param name A name of a repository.
+   * @param github An interface to the GitHub API.
+   * @param githubToken provided by user to access GitHub API.
+   */
+  public UsesSignedCommits(String where, String name, GitHub github, String githubToken)
+      throws IOException {
+    super(where, name, github);
+    this.githubToken = githubToken;
+  }
+
+  @Override
+  public UsesSignedCommits update(ValueSet values) throws IOException {
+    Objects.requireNonNull(values, "Hey! Values can't be null!");
+    System.out.println("[+] Figuring out if the project uses verified signed commits ...");
+
+    Optional<Value> something = cache().get(url, USES_VERIFIED_SIGNED_COMMITS);
+    if (something.isPresent()) {
+      values.update(something.get());
+      return this;
+    }
+
+    Value<Boolean> signedCommits = usesSignedCommits();
+    values.update(signedCommits);
+    cache().put(url, signedCommits, tomorrow());
+
+    return this;
+  }
+
+  /**
+   * It gathers the list of commits within one year from the current date. Then it calculates
+   * density ratio. The density ratio is the percentage calculation of number of verified signed
+   * commits against the total number of commits in the year. If the density ratio in % crosses a
+   * certain percentage threshold ({@link #SIGNED_COMMIT_PERCENTAGE_THRESHOLD}), then it says that
+   * the project uses verified signed commits. Otherwise, it says that the project does not use
+   * verified signed commits.
+   * 
+   * @return {@link BooleanValue} value of the feature {@link #USES_VERIFIED_SIGNED_COMMITS}.
+   */
+  private Value<Boolean> usesSignedCommits() {
+    double signedCounter = 0;
+    double counter = 0;
+    Date date = new Date(System.currentTimeMillis() - DELTA);
+    String commitDate = DATE_FORMAT.format(date);
+    int pageNumber = FIRST_PAGE;
+    boolean hasNextPage = true;
+
+    // TODO: The data gathering process here from GitHub API will have to be merged with other
+    // related data providers. More information can be found in
+    // https://github.com/SAP/fosstars-rating-core/issues/69
+    try {
+      while (hasNextPage) {
+        CloseableHttpResponse response = commitInfo(commitDate, pageNumber);
+        JsonNode commitNodeList = MAPPER.readTree(response.getEntity().getContent());
+        System.out.println(String.format("Page %s, Count %s", pageNumber, commitNodeList.size()));
+        if (commitNodeList == null || commitNodeList.size() == 0) {
+          break;
+        }
+
+        hasNextPage = hasNextPage(response);
```

Comment:
One minor thing. The variable doesn't seem to be really necessary. How about using a do-while loop to make it a bit shorter:

```java
do {
    CloseableHttpResponse response = commitInfo(commitDate, pageNumber);
    // do stuff
} while (hasNextPage(response));
```